nil/core-concepts/shards-parallel-execution.mdx

```diff
@@ -29,6 +29,92 @@ As the number of execution shards grows, this process may be changed so that sha
 
 :::
 
+### Async execution
+
+Generally, the term 'async execution' describes delegating the completion of a task to a dedicated CPU thread or a similar isolated component. While this task is processed, the main program can freely execute other instructions, and then react to receiving the results of the async task.
+
+In =nil;, async execution refers to calling a contract that is deployed on another shard. The time taken to execute this call may vary, and the 'caller' contract is also free to execute other instructions without having to wait for the results of the call.
+
+Consider two contracts deployed on two different shards in =nil;:
+
+* Contract 1 (caller) is deployed on Shard 1
+* Contract 2 (receiver) is deployed on Shard 2
+
+The below diagram displays how an async call could work between Contract 1 and Contract 2.
+
+```mermaid
+%%{
+  init: {
+    'theme': 'base',
+    'themeVariables': {
+      'primaryColor': '#181A1B',
+      'primaryTextColor': '#f1f1f1',
+      'primaryBorderColor': '#434343',
+      'lineColor': '#F8B229',
+      'secondaryColor': '#181A1B',
+      'tertiaryColor': '#fff'
+    }
+  }
+}%%
+sequenceDiagram
+    participant S1 as Shard 1 with Contract 1
+    participant S2 as Shard 2 with Contract 2
+    S1 ->> S1: Contract 1 calls a function<br>inside Contract 2
+    S1 ->> S1: Shard 1 creates a new transaction<br>whose destination is Shard 2
+    Note right of S1: Contract 1 is freed and can execute other operations
+    S1 --> S2: Shard 2 picks up the transaction<br>and adds it to the mempool
+    S2 ->> S2: Shard 2 executes the transaction<br>and includes it in a block
+    S2 ->> S2: Contract 2 executes the function<br>that Contract 1 called   
+```
+
+The key stages of this flow are:
+
+1. Contract 1 calls a function inside Contract 2, which creates a new transaction
+2. Shard 2 must pick up this transaction and place it inside its mempool
```

Comment:
Using the term "mempool" here is factually incorrect and may lead to impression that messages can disappear. 

What happens is a bit different:
- Shard 1 persists an outgoing message in a block
- Shard 2 goes into creation of new block
  - It picks some messages from the mempool (external)
  - And picks messages from other shards (internal)
- It then creates a new block and processes messages one by one

The way it picks up messages from other shards - it listens for their latest blocks and copies up to a certain number of messages that have it as a destination.

Reply:
Thank you, I have changed the diagram and added more info about copying messages